pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,580 @@
+			reqLogger.Info("Service was just created, so let's try to requeue to set it up")
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		} else {
+			return reconcile.Result{}, err
+		}
+	} else {
+		// Service exists, check if it's updated.
+		// XXX Copy system-assigned fields to satisfy reflect.DeepEqual.
+		service.Spec.Ports[0].NodePort = foundService.Spec.Ports[0].NodePort
+		service.Spec.ClusterIP = foundService.Spec.ClusterIP
+		service.Spec.HealthCheckNodePort = foundService.Spec.HealthCheckNodePort
+		if !reflect.DeepEqual(foundService.Spec, service.Spec) {
+			// Specs aren't equal, update and fix.
+			r.SetSSHDStatusPending(instance, "Updating service", "from", foundService.Spec, "to", service.Spec)
+			foundService.Spec = *service.Spec.DeepCopy()
+			if err = r.client.Update(context.TODO(), foundService); err != nil {
+				r.SetSSHDStatusError(instance, "Failed to update service", err)
+				return reconcile.Result{}, err
+			}
+			// Requeue to give AWS time to apply the changes.
+			reqLogger.Info("Requeuing after service update")
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+	}
+
+	// Create a Route53 DNS entry for the service's load balancer.
+	// TODO Consider using https://github.com/openshift/external-dns
+
+	clusterBaseDomain, err := utils.GetClusterBaseDomain(r.client)
+	if err != nil {
+		r.SetSSHDStatusError(instance, "Failed to get cluster's base domain", err)
+		return reconcile.Result{}, err
+	}
+
+	// Get the ELB-to-be's name from Service's UID.
+	elbName := strings.ReplaceAll("a"+string(foundService.ObjectMeta.UID), "-", "")
+	if len(elbName) > 32 {
+		// truncate to 32 characters
+		elbName = elbName[0:32]
+	}
+
+	exists, elb, err := r.awsClient.DoesELBExist(elbName)
+	if err != nil {
+		r.SetSSHDStatusError(instance, "Failed to get load balancer status", err)
+		return reconcile.Result{}, err
+	}
+	if !exists {
+		// It isn't bad that it doesn't exist if there's no error, so re-queue.
+		r.SetSSHDStatusPending(instance, "Waiting on service load balancers")
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	r.route53 = &Route53Data{
+		loadBalancerDNSName:      elb.DNSName,
+		loadBalancerHostedZoneId: elb.DNSZoneId,
+		resourceRecordSetName:    instance.Spec.DNSName + "." + clusterBaseDomain,
+		privateHostedZoneName:    clusterBaseDomain + ".",
+		// The public zone name omits the cluster name.
+		// e.g. mycluster.abcd.s1.openshift.com -> abcd.s1.openshift.com
+		publicHostedZoneName: clusterBaseDomain[strings.Index(clusterBaseDomain, ".")+1:] + ".",
+	}
+
+	err = r.ensureDNSRecords()
+	if err != nil {
+		r.SetSSHDStatusError(instance, "Failed to ensure the DNS record", err)
+		return reconcile.Result{}, err
+	}
+
+	r.SetSSHDStatus(instance, "SSHD is ready", cloudingressv1alpha1.SSHDStateReady)
+
+	return reconcile.Result{RequeueAfter: 60 * time.Second}, nil
```

Comment:
I'm not sure you need to reconcile every minute like the apischeme controller because your dependent watches should cause reconciliation upon their change.